hypothesis-python/src/hypothesis/database.py

```diff
@@ -568,11 +599,16 @@ def _fetch_artifact(self) -> Optional[Path]:  # pragma: no cover
         response_bytes = self._get_bytes(url)
         if response_bytes is None:
             return None
+
         artifacts = json.loads(response_bytes)["artifacts"]
+        artifacts = filter(lambda a: a["name"] == self.artifact_name, artifacts)
+
+        if not artifacts:
+            return None
 
         # Get the latest artifact from the list
         artifact = sorted(
-            filter(lambda a: a["name"] == self.artifact_name, artifacts),
+            artifacts,
             key=lambda a: a["created_at"],
         )[-1]
```

Comment:
Here, I'd prefer `artifact = max(artifacts, key=lambda a: a["created_at"])` - no need to order all the elements we don't care about!